File: nodejs/src/client/review.ts
/**
 * Review facade (MCP: loxtep_review).
 * Delegates to approvals, improvements, CDLC, and context-mining APIs.
 */

import type { createApprovalsApi } from './approvals.js';
import type { CdlcApi } from './cdlc.js';
import type { ImprovementsApi } from './improvements.js';
import type { MiningApi } from './mining.js';

export interface ReviewFacadeDeps {
  approvals: ReturnType<typeof createApprovalsApi>;
  improvements: ImprovementsApi;
  cdlc: CdlcApi;
  mining: MiningApi;
}

export function createReviewFacade(deps: ReviewFacadeDeps): {
  approvals: ReviewFacadeDeps['approvals'];
  improvements: ReviewFacadeDeps['improvements'];
  cdlc: ReviewFacadeDeps['cdlc'];
  mining: ReviewFacadeDeps['mining'];
} {
  return {
    approvals: deps.approvals,
    improvements: deps.improvements,
    cdlc: deps.cdlc,
    mining: deps.mining,
  };
}

export type ReviewFacade = ReturnType<typeof createReviewFacade>;

File: python/src/loxtep/review.py
"""Review facade (MCP: loxtep_review)."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Union

from .approvals import ApprovalsApi, AsyncApprovalsApi
from .improvements import AsyncImprovementsApi, ImprovementsApi


@dataclass(frozen=True)
class ReviewFacade:
    approvals: Union[ApprovalsApi, AsyncApprovalsApi]
    improvements: Union[ImprovementsApi, AsyncImprovementsApi]
